src/or_trajopt/planning.py

```diff
@@ -164,7 +168,7 @@ def _addFunction(problem, timestep, i_dofs, n_dofs, fnargs):
             ValueError('Invalid cost or constraint type: {:s}'
                        .format(str(fntype)))
 
-    def _Plan(self, robot, request,
+    def _Plan(self, robot, robot_checker, request,
```

Comment:
`self.robot_collision_checker` is a factory that creates a `robot_checker` context manager. This additional level of indirection allows us to do baking in `__enter__` and cleanup baking in `__exit__`. It also lets us wrap the `__enter__` and `__exit__` methods of a `CollisionOptionsStateSaver` to enable `CollisionOptions.ActiveDOFs`.

All of the internal methods take `robot_checker` so we only do the baking *once* per query. The usage looks pattern looks like this:
```python
with self.robot_collision_checker(robot) as robot_checker:
    return self._Plan(robot, robot_checker, request, **kwargs)
```

The name `robot_collision_checker` and `robot_checker` is confusing, but is chosen to be consistent with the other planners in PrPy. @cdellin and I discussed this in https://github.com/personalrobotics/prpy/pull/343, but did not come up with a better name. I would prefer to stick with these names for now and refactor everything (including all of the planners in PrPy) at once.